### PR TITLE
fix: display correct reference field when it is a smartfield

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: test/services/smart-fields-values-injector.test.js [hunk 1]
```diff
@@ -100,4 +100,18 @@ describe('services > smart-fields-values-injector', () => {
       expect(addressRecord.smartUser.hasAddress).toBe(true);
     });
   });
+
+  describe('with no fieldsPerModel', () => {
+    const fieldsPerModel = undefined;
+    it('should inject the Smart Relationship reference', async () => {
```
Comment (VincentMolinie) [inline test/services/smart-fields-values-injector.test.js:106]: [Code Style] A new line before this line

**Review by VincentMolinie — APPROVED** [review]
Good job !

**Review by SteveBunlon — APPROVED** [review]
Tested it and could not reproduce the issue ✅ 

I've perform some more tests to ensure nothing else has been impacted and I could not find anything. 

Let's 🔥

File: src/services/smart-fields-values-injector.js [hunk 2]
```diff
@@ -76,10 +76,11 @@ function SmartFieldsValuesInjector(
 
   this.perform = () =>
     P.each(schema.fields, (field) => {
-      if (record
-          && field.isVirtual
-          && (field.get || field.value)
-          && isRequestedField(requestedField || modelName, field.field)) {
+      if (record && field.isVirtual && (field.get || field.value)) {
+        if (fieldsPerModel && !isRequestedField(requestedField || modelName, field.field)) {
```
[the remark this answers is not in the document]
Comment (GuillaumeCisco) [reply src/services/smart-fields-values-injector.js:80]: `fieldsPerModel` represent the fields you want to display (can include smartFields). Used when filtering.
I have to `fieldsPerModel` in the condition as the file is not architectured as a classic js class...
I think refactoring this whole file and process would be great in the future :)

**Pull request merged**